<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>640</width>
    <height>480</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>640</width>
    <height>480</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>640</width>
    <height>480</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Password Generator</string>
  </property>
  <property name="windowIcon">
   <iconset>
    <normaloff>:/icons/resources/app_icon.ico</normaloff>:/icons/resources/app_icon.ico</iconset>
  </property>
  <widget class="QWidget" name="centralwidget">
   <property name="styleSheet">
    <string notr="true">QWidget {
    background-color: #ececec;
    color: #142d4c;
    font-family: Roboto;
    font-size: 16pt;
    margin: 10px;
}

QPushButton {
    border: 2px solid #385170;
	padding: 10px;
    border-radius: 5px;
	margin: 0 5px 0 0;
}

QPushButton#btn_lower, #btn_upper, #btn_digits, #btn_punct{
    padding: 10px 0;
	margin-left: 5px;
}

QPushButton:hover {
    border-color: #9fd3c7;
}

QPushButton:pressed {
    border: 6px solid #385170;
}

QPushButton:checked {
    background-color: #9fd3c7;
    border: 4px solid #385170;
    border-radius: 5px;
}

QLineEdit {
    border: 2px solid #385170;
	border-radius: 5px;
    margin-left: 5px;
    font-size: 20pt;
}

QSlider::groove:horizontal {
    background-color: transparent;
    height: 5px;
}

QSlider::sub-page:horizontal {
    background-color: #9fd3c7;
}

QSlider::add-page:horizontal {
    background-color: #385170;
}

QSlider::handle:horizontal {
    background-color: #ececec;
	border: 2px solid #385170;
    width: 18px;
    border-radius: 10px;
    margin-top: -8px;
    margin-bottom: -8px;
}

QSpinBox {
    background: transparent;
    border: 2px solid #385170;
	padding: 10px;
    border-radius: 5px;
}

QSpinBox:hover {
    border-color: #9fd3c7;
}</string>
   </property>
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <widget class="QLineEdit" name="line_password">
        <property name="readOnly">
         <bool>true</bool>
        </property>
        <property name="placeholderText">
         <string>Generated password</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="btn_refresh">
        <property name="text">
         <string/>
        </property>
        <property name="icon">
         <iconset>
          <normalon>:/icons/resources/autorenew_black_24dp.svg</normalon>
         </iconset>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="btn_copy">
        <property name="text">
         <string/>
        </property>
        <property name="icon">
         <iconset>
          <normalon>:/icons/resources/content_copy_black_24dp.svg</normalon>
         </iconset>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QHBoxLayout" name="layout_info">
      <item>
       <widget class="QLabel" name="label_entropy">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="text">
         <string/>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="label_strength">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="text">
         <string/>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QHBoxLayout" name="layout_length">
      <item>
       <widget class="QLabel" name="label">
        <property name="text">
         <string>Password length:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSlider" name="slider_length">
        <property name="cursor">
         <cursorShape>PointingHandCursor</cursorShape>
        </property>
        <property name="value">
         <number>15</number>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSpinBox" name="spinbox_length">
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
        <property name="buttonSymbols">
         <enum>QAbstractSpinBox::NoButtons</enum>
        </property>
        <property name="value">
         <number>15</number>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QHBoxLayout" name="layout_characters">
      <item>
       <widget class="QLabel" name="label_2">
        <property name="text">
         <string>Include:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="btn_lower">
        <property name="text">
         <string>[a-z]</string>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="btn_upper">
        <property name="text">
         <string>[A-Z]</string>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="btn_digits">
        <property name="text">
         <string>[0-9]</string>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="btn_punct">
        <property name="text">
         <string>[#@!$%...]</string>
        </property>
        <property name="checkable">
         <bool>true</bool>
        </property>
        <property name="checked">
         <bool>false</bool>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
 </widget>
 <resources>
  <include location="../resources.qrc"/>
 </resources>
 <connections/>
</ui>
